Port DropBox to the com.endlessm.apps.Platform

mariospr opened the pull request:
https://phabricator.endlessm.com/T22890

Commit:
Remove autotools & debian related files from the repository

We're going to be building Dropbox as a regular flatpak with flatpak-builder
from now on, using the freedesktop runtime, so there's no need to keep these
files around anymore.

https://phabricator.endlessm.com/T22890

Commit:
Move everything out of src/, into the top directory

It does not make sense anymore to have this artificial separation
and will make the JSON manifest easier and cleaner, so move it.

https://phabricator.endlessm.com/T22890

Commit:
Add .desktop file, appdata information and icon.

All taken from our current Dropbox package.

https://phabricator.endlessm.com/T22890

Commit:
Add a native JSON manifest to build Dropbox

Ideally, we'd build it against the Freedesktop runtime, but the
launcher scripts rely on GNOME stuff, so we'll need to depend at
least against org.gnome.Platform unless we want to reimplement
the logic not to depend o that, which is out of scope for now.

https://phabricator.endlessm.com/T22890

Commit:
Remove our version of xdg-open, no longer needed

This was added to deal with situations where the user might have
installed either an older version (pre 0.6) of the portals or
the new one, to gracefully deal with OpenFile or OpenURI methods,
as well as to make sure the OpenURI portal was being used when
xdg-open was invoked from the Dropbox daemon.

Since we're using now a newer runtime which ships a flatpak-specific
version of xdg-open that interacts correctly with the portal, and
the minimum version of the portals users in 3.4+ versions of EOS
will be good enough, there's no need to keep this around so drop it.

https://phabricator.endlessm.com/T22890

nedrichards commented on com.dropbox.Client.json:
```diff
@@ -0,0 +1,116 @@
+{
+    "app-id": "com.dropbox.Client",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
```
Why not 3.28?

mariospr replied:
No good reason. I think my brain picked 3.26 for consistency with our desktop, but that's total nonsense now that 3.28 is stable. I'll change that

nedrichards commented on com.dropbox.Client.json:
```diff
@@ -0,0 +1,116 @@
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* Sound and network access */
+        "--socket=pulseaudio",
```
What does dropbox need sound for?

mariospr replied:
I have no clue, really, but I decided to keep it in case there was some "beep" sound somewhere I was missing (I don't really use Dropbox myself much, so I did err on the "safe" side here).

I can remove it if you want / feel confident there should not be any sound coming out of this.

ramcq commented on portallauncher.py:
```diff
@@ -27,7 +27,6 @@
 import errno
 import logging
 import os
-import sys
```
This renaming patch appears to contain a code change? It would ideally be committed separately.

mariospr replied:
It's an unused import, I thought it was not relevant enough to grant it a separate commit, but I can do that, sure.

mariospr replied:
Not relevant anymore: the new PR won't include this file at all

ramcq commented on com.dropbox.Client.appdata.xml:
```diff
@@ -0,0 +1,76 @@
+
+<metadata>
+  <value key="EndlessOS::legacy-ext-runtime-installed">d8aee708de68406b036f323f0ad06f0f4353c133328d471b84d2ff9a52ae30da</value>
+  <value key="EndlessOS::available-since">3.1</value>
```
I think these two can be removed safely now. They are to do with the move from the old "endless extra data" apps in 3.0 to the upstream solution in 3.1.

mariospr replied:
Oops! I removed those from the json file but forgot these ones. Will remove them

ramcq commented on com.dropbox.Client.desktop:
```diff
@@ -0,0 +1,36 @@
+Icon=com.dropbox.Client
+Categories=Utility;
+X-Endless-LaunchMaximized=false
+X-Flatpak=com.dropbox.Client
```
This can be removed, it's added by Flatpak on export.

mariospr replied:
Ok

ramcq commented on eos-dropbox-app:
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+prefix=/app
+exec /app/share/eos-dropbox-app/eos-dropbox-app.py "$@"
```
Can this entire script just be removed if we make the eos-dropbox-app.py +x and just point the desktop file to it directly?

ramcq replied:
Oh I guess actually it lets us set the flatpak command. I guess we can leave it, it's harmless. :)

mariospr replied:
Actually, it WILL be removed... this morning I realized I don't need the portallauncher.py at all and so we can simplify things a lot now that neither that file nor xdg-open.py are included: stop shipping things in `share/eos-dropbox-app` and simply make `/app/bin/eos-dropbox-app` the python script.

Will include this change in the reviewed PR

ramcq commented on com.dropbox.Client.json:
```diff
@@ -0,0 +1,116 @@
+{
+    "app-id": "com.dropbox.Client",
+    "runtime": "org.gnome.Platform",
```
GitHub doesn't let you comment commit messages, but the commit message needs rewording here to match the decision to use org.gnome.Platform in the end.

mariospr replied:
Sure

ramcq reviewed (changes requested):
A few minor things.

Commit:
Remove the eos-dropbox-app launcher script

We will only be installing one python script now and there's no need
to install the .py files in one place and provide a launcher (we will
install eos-dropbox-app.py as /app/bin/eos-dropbox-app), so we can
simplify this further now as well.

https://phabricator.endlessm.com/T22890

Commit:
Remove portallauncher module, we can finally rely on GLib from now on

Similar	to why we removed xdg.open.py, we don't	need to	consider OpenURI
and OpenFile anymore to	open the Dropbox directory, making this module no
longer needed (we can run g_app_info_launch_default_for_uri), so drop it.

https://phabricator.endlessm.com/T22890
